jax_galsim/interpolatedimage.py

```diff
@@ -0,0 +1,1115 @@
+
+    @lazy_property
+    def _xim(self):
+        pad_factor = self._jax_aux_data["pad_factor"]
+
+        # The size of the final padded image is the largest of the various size specifications
+        pad_size = max(self._image.array.shape)
+        if pad_factor > 1.0:
+            pad_size = int(math.ceil(pad_factor * pad_size))
+        # And round up to a good fft size
+        pad_size = Image.good_fft_size(pad_size)
+
+        xim = Image(
+            _zeropad_image(
+                self._image.array, (pad_size - max(self._image.array.shape)) // 2
+            ),
+            wcs=PixelScale(1.0),
+        )
+        xim.setCenter(0, 0)
+        xim.wcs = PixelScale(1.0)
+
+        # Now place the given image in the center of the padding image:
+        # assert self._xim.bounds.includes(self._image.bounds)
+        xim[self._image.bounds] = self._image
+
+        return xim
+
+    @lazy_property
+    def _pad_image(self):
+        # These next two allow for easy pickling/repring.  We don't need to serialize all the
+        # zeros around the edge.  But we do need to keep any non-zero padding as a pad_image.
+        xim = self._xim
+        nz_bounds = self._image.bounds
+        return xim[nz_bounds]
+
+    @lazy_property
+    def _kim(self):
+        return self._xim.calculate_fft()
+
+    @lazy_property
+    def _pos_neg_fluxes(self):
+        # record pos and neg fluxes now too
+        pflux = jnp.sum(jnp.where(self._pad_image.array > 0, self._pad_image.array, 0))
+        nflux = jnp.abs(
+            jnp.sum(jnp.where(self._pad_image.array < 0, self._pad_image.array, 0))
+        )
+        pint = self._x_interpolant.positive_flux
+        nint = self._x_interpolant.negative_flux
+        pint2d = pint * pint + nint * nint
+        nint2d = 2 * pint * nint
+        return [
+            pint2d * pflux + nint2d * nflux,
+            pint2d * nflux + nint2d * pflux,
+        ]
+
+    @property
+    def _positive_flux(self):
+        return self._pos_neg_fluxes[0]
+
+    @property
+    def _negative_flux(self):
+        return self._pos_neg_fluxes[1]
+
+    def _flux_per_photon(self):
+        return self._calculate_flux_per_photon()
+
+    @lazy_property
+    def _maxk(self):
+        if self._jax_aux_data["_force_maxk"]:
+            major, minor = compute_major_minor_from_jacobian(
+                self._jac_arr.reshape((2, 2))
+            )
+            return self._jax_aux_data["_force_maxk"] * minor
+        else:
+            return self._getMaxK(self._jax_aux_data["calculate_maxk"])
+
+    @lazy_property
+    def _stepk(self):
+        if self._jax_aux_data["_force_stepk"]:
+            major, minor = compute_major_minor_from_jacobian(
+                self._jac_arr.reshape((2, 2))
+            )
+            return self._jax_aux_data["_force_stepk"] * minor
+        else:
+            return self._getStepK(self._jax_aux_data["calculate_stepk"])
+
+    def _getStepK(self, calculate_stepk):
```

Comment:
why not include the `_force_stepk` argument like in Galsim?

Reply:
So force stepk and force maxk are handled differently than galsim. There are branches in the properties above that enforce them and also scale them properly to account for the jacobian.